internal/controller/apiextensions/composite/composition_ptf.go

```diff
@@ -649,7 +649,7 @@ type ContainerFunctionRunnerOption func(ctx context.Context, fn *v1.ContainerFun
 // 2. Loads credentials from the supplied service account's image pull secrets.
 // 3. Loads credentials from the function's image pull secrets.
 // 4. Loads credentials using the GKE, EKS, or AKS credentials helper.
-func WithKubernetesAuthentication(c client.Reader, namespace, serviceAccount string) ContainerFunctionRunnerOption {
+func WithKubernetesAuthentication(c client.Reader, namespace, serviceAccount string, registry string) ContainerFunctionRunnerOption {
```

Comment:
```suggestion
func WithKubernetesAuthentication(c client.Reader, namespace, serviceAccount, registry string) ContainerFunctionRunnerOption {
```

Nit: In Go you can "dedupe" the type when you have several arguments of the same type in a row. 🙂

Reply:
Can you fix this up and then I think I'm ready to merge.